docker/auth/auth.py

```diff
@@ -102,12 +102,13 @@ def encode_header(auth):
     return base64.urlsafe_b64encode(auth_json)
 
 
-def parse_auth(entries):
+def parse_auth(entries, fail=False):
```

Comment:
Minor: I think the kwarg could be slightly more verbose, maybe `warn_on_error=True` or `raise_on_error=False` ?